hibernate-core/src/main/java/org/hibernate/proxy/AbstractLazyInitializer.java

```diff
@@ -173,6 +188,7 @@ else if ( !session.isConnected() ) {
 				else {
 					target = session.immediateLoad( entityName, id );
```

Comment:
Does the change you did in `SessionImpl#getCurrentLockMode` affect the SQL statement to include the lock mode in the select statement here? It should IMO, and if it does, then you shouldn't have to set the lock mode on the entity entry afterwards AFAIU.

Reply:
the select statement performed by the `immediateLoad` does not include the lock